File: TTTChatRoom/TTTChatRoom/TCRManager.m
//
//  TCRManager.m
//  TTTChatRoom
//
//  Created by Work on 2019/1/18.
//  Copyright © 2019 Work. All rights reserved.
//

#import "TCRManager.h"

@implementation TCRManager
static id _manager;
+ (instancetype)manager
{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [[self alloc] init];
    });
    return _manager;
}

+(instancetype)allocWithZone:(struct _NSZone *)zone
{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [super allocWithZone:zone];
    });
    return _manager;
}

- (id)copyWithZone:(NSZone *)zone
{
    return _manager;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        _rtcEngine = [TTTRtcEngineKit sharedEngineWithAppId:@"496e737d22ecccb8cfa780406b9964d0" delegate:nil];
        _me = [[TCRUser alloc] initWithUid:0];
    }
    return self;
}
@end

File: ChatRoom-Swift/TTTChatRoom/TCRManager.swift
//
//  TCRManager.swift
//  TTTChatRoom
//
//  Created by Work on 2019/3/13.
//  Copyright © 2019 yanzhen. All rights reserved.
//

import UIKit
import TTTRtcEngineVoiceKit

let TTManager = TCRManager.manager

class TCRManager: NSObject {

    public static let manager = TCRManager()
    public var rtcEngine: TTTRtcEngineKit!
    public var me = TCRUser(0)
    
    private override init() {
        super.init()
        //输入对应的AppId
        rtcEngine = TTTRtcEngineKit.sharedEngine(withAppId: <#name#>, delegate: nil)
    }
}
